### ISSUE TITLE
no-invalid-regexp: Validate flags even for non-literal patterns

### ISSUE THREAD

**Issue by trosos** [eslint/eslint#16573]
Hi,

The `no-invalid-regexp` rule skips flag validation for regexps with non-literal pattern.
I believe that flags should be validated no matter what the pattern is.

### What rule do you want to change?

no-invalid-regexp

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
RegExp("." + "", "a");
```

### What does the rule currently do for this code?

The rule skips flag validation and reports no issues.

### What will the rule do after it's changed?

The rule should IMO report invalid flag.

### Participation

- [X] I am willing to submit a pull request to implement this change.

**Comment by mdjermanovic** [comment]
Hi @trosos, thanks for the issue!

 > The `no-invalid-regexp` rule skips flag validation for regexps with non-literal pattern.
> I believe that flags should be validated no matter what the pattern is.

Makes sense, PR is welcome! I think we can treat this as a semver-minor bug fix.

**Comment by trosos** [comment]
Thank you, @mdjermanovic, for confirmation.
I am working on a fix right now.
